app/services/authn_context_resolver.rb

```diff
@@ -10,11 +10,24 @@ def initialize(user:, service_provider:, vtr:, acr_values:)
     @acr_values = acr_values
   end
 
-  def resolve
-    if vtr.present?
-      selected_vtr_parser_result_from_vtr_list
+  def result
+    @result ||= if vtr.present?
```

Comment:
hm,  @jmhooper were there situations where we wouldn't want to memoize the actual result, vs memozing in the caller class? i'm a little nervous about this change affecting something unexpectedly.

Reply:
In all the callers of `AuthContextResolver`, I see the result being memoized:

1. app/controllers/application_controller.rb
2. app/presenters/openid_connect_user_info_presenter.rb
3. app/services/analytics.rb
4. app/services/attribute_asserter.rb
5. app/services/authn_context_resolver.rb
6. app/services/id_token_builder.rb

Reply:
Yeah, I think this memoization should be safe